tests/unit/test_operator.py

```diff
@@ -60,3 +60,43 @@ def test_with_relation(harness):
 
     _ = harness.get_pod_spec()
     assert isinstance(harness.charm.model.unit.status, ActiveStatus)
+
+
+def test_public_url_prepend_http(harness):
+    harness.set_leader(True)
+    harness.add_oci_resource(
+        "oci-image",
+        {
+            "registrypath": "ci-test",
+            "username": "",
+            "password": "",
+        },
+    )
+    harness.update_config({"public-url": "10.64.140.43.nip.io"})
+    harness.begin_with_initial_hooks()
+
+    pod_spec, _ = harness.get_pod_spec()
+
+    assert pod_spec["containers"][0]["envConfig"]["OIDC_PROVIDER"].startswith(
```

Comment:
My first reaction is that this should be an equality check rather than `startswith` (that it is exactly `http://10..../dex`), but that's also very picky.  In this case you know exactly what the value should be so it seems more appropriate to test I think, but if you disagree leave this as is.  Same as the other test below